server/core/src/rsApiHandler.cpp

```diff
@@ -233,63 +234,58 @@ int rsApiHandler(rsComm_t*   rsComm,
                      myArgv[3]);
     }
 
-    if ( retVal != SYS_NO_HANDLER_REPLY_MSG ) {
-        status = sendAndProcApiReply
-                 ( rsComm, apiInx, retVal, myOutStruct, &myOutBsBBuf );
+    if (retVal != SYS_NO_HANDLER_REPLY_MSG) {
+        status = sendAndProcApiReply(rsComm, apiInx, retVal, myOutStruct, &myOutBsBBuf);
     }
 
-    // =-=-=-=-=-=-=-
     // clear the incoming packing instruction
-    if ( myInStruct != NULL ) {
-        if ( RsApiTable[apiInx]->clearInStruct ) {
-            RsApiTable[apiInx]->clearInStruct( myInStruct );
+    if (myInStruct) {
+        if (RsApiTable[apiInx]->clearInStruct) {
+            RsApiTable[apiInx]->clearInStruct(myInStruct);
         }
 
-        free( myInStruct );
-        myInStruct = NULL;
+        std::free(myInStruct);
+        myInStruct = nullptr;
     }
 
-    if ( retVal >= 0 && status < 0 ) {
+    if (retVal >= 0 && status < 0) {
         return status;
     }
-    else {
-        return retVal;
-    }
+
+    return retVal;
 }
 
-int
-sendAndProcApiReply( rsComm_t * rsComm, int apiInx, int status,
-                     void * myOutStruct, bytesBuf_t * myOutBsBBuf ) {
-    int retval;
+int sendAndProcApiReply(rsComm_t* rsComm, int apiInx, int status, void*& myOutStruct, bytesBuf_t* myOutBsBBuf)
```

Comment:
Notice `void*&`.

This allows other function calls to see when `myOutStruct` is changed (set to nullptr).

Reply:
This is a good thing.